core/sql/sqlcomp/PrivMgrMD.h

```diff
@@ -110,12 +106,35 @@ typedef struct {
 
 } ViewUsage;
 
-typedef struct {
+// -----------------------------------------------------------------------
+// Class definitions
+// -----------------------------------------------------------------------
+class ObjectReference
+{
+  public:
+
+  ObjectReference()
+  : objectUID(0),
+    objectOwner(NA_UserIdDefault),
+    objectType(COM_UNKNOWN_OBJECT),
+    columnReferences(NULL),
+    updatedPrivs()
+  {}
+    
+  virtual ~ObjectReference ( void )
+  {
+    while(!columnReferences->empty())
+      delete columnReferences->back(), columnReferences->pop_back();
```

Comment:
I was having issues getting this code to work and I searched the web for examples.  This was one suggestion and it worked.